templates/csharp/app-reactiveui/ViewLocator.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using Avalonia.Controls;
+using Avalonia.Controls.Templates;
+using AvaloniaAppTemplate.ViewModels;
+
+namespace AvaloniaAppTemplate
+{
+    public class ViewLocator : IDataTemplate
```

Comment:
ReactiveUI provides its own view location system which includes `IViewLocator` and friends such as `RoutingState` and `RoutedViewHost`, so probably worth including some basic navigation examples into the template instead of this `ViewLocator`. The `RoutedViewHost` is usually placed in the root window XAML:
```xaml
<reactiveUi:RoutedViewHost Router="{Binding Router}">
    <reactiveUi:RoutedViewHost.PageTransition>
        <CrossFade Duration="0.15" />
    </reactiveUi:RoutedViewHost.PageTransition>
    <reactiveUi:RoutedViewHost.ViewLocator>
        <sourceFileEditor:SourceFileEditorLocator />
    </reactiveUi:RoutedViewHost.ViewLocator>
</reactiveUi:RoutedViewHost>
```
And a simple `IViewLocator` is given by (here the *View classes are `UserControl`s):
```
public sealed class SourceFileEditorLocator : IViewLocator
{
    public IViewFor ResolveView<T>(T viewModel, string contract = null) where T : class => viewModel switch
    {
        SourceFileEditorFileViewModel file => new SourceFileEditorFileView(),
        SourceFileOverlayViewModel editor => new SourceFileOverlayView(),
        SourceFileEditorSwitchViewModel tabs => new SourceFileEditorSwitchView(),
    };
}
```

Reply:
I think work with routing can be added with an additional example 😉